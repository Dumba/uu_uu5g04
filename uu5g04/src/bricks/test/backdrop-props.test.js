/**
 * Copyright (C) 2019 Unicorn a.s.
 * 
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 * 
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

import React from "react";
import UU5 from "uu5g04";
import "uu5g04-bricks";
import createReactClass from "create-react-class";

const { mount, shallow, wait } = UU5.Test.Tools;

const MyBackDropComponent = createReactClass({

  getInitialState: () => {
    return {
      isCalled: false
    };
  },

  onClickAlert(event) {
    alert("You just clicked on the backdrop");
    this.setState({isCalled: true})
  },

  onMouseOverAlert(event) {
    alert("You have mouse over on the backdrop");
    this.setState({isCalled: true})
  },

  render() {
    return (
      <UU5.Bricks.Backdrop
        id={"idBackDrop"}
        onClick={this.onClickAlert}
        onMouseOver={this.onMouseOverAlert}
      />
    );
  }
});

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.NestingLevelMixin",
    "UU5.Common.PureRenderMixin"
  ],
  props: {
    background: {
      values: [true, false]
    }
    //onClick:{}
    //onMouseOver:{}
  },
  requiredProps: {
    children: [<UU5.Bricks.Jumbotron content={"JumboTron Content"} className="center" id={"jumboID"}/>]
  },
  opt: {
  }
};


describe(`UU5.Bricks.Backdrop`, () => {
  UU5.Test.Tools.testProperties(UU5.Bricks.Backdrop, CONFIG);
});

describe(`UU5.Bricks.Backdrop  props.function `, () => {
  it(`UU5.Bricks.Backdrop - onClick()`, () => {
    window.alert = jest.fn();
    const wrapper = shallow(<MyBackDropComponent/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().isCalled).toBeFalsy();
    wrapper.simulate('click');
    expect(window.alert).toBeCalled();
    expect(window.alert).toHaveBeenCalledWith('You just clicked on the backdrop');
    expect(wrapper.state().isCalled).toBeTruthy();
    expect(window.alert.mock.calls[0][0]).toEqual("You just clicked on the backdrop");
    expect(wrapper).toMatchSnapshot();
  });

  it(`UU5.Bricks.Backdrop - onMouseOver()`, () => {
    window.alert = jest.fn();
    const wrapper = shallow(<MyBackDropComponent/>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().isCalled).toBeFalsy();
    wrapper.simulate('mouseOver');
    expect(window.alert).toBeCalled();
    expect(window.alert).toHaveBeenCalledWith('You have mouse over on the backdrop');
    expect(wrapper.state().isCalled).toBeTruthy();
    expect(window.alert.mock.calls[0][0]).toEqual("You have mouse over on the backdrop");
    expect(wrapper).toMatchSnapshot();
  });
});
