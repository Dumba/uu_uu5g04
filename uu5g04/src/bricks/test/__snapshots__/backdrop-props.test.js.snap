// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Backdrop  props.function  UU5.Bricks.Backdrop - onClick() 1`] = `
<Backdrop
  background={false}
  className={null}
  controlled={true}
  disabled={false}
  hidden={false}
  id="idBackDrop"
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  onClick={[Function]}
  onMouseOver={[Function]}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
/>
`;

exports[`UU5.Bricks.Backdrop  props.function  UU5.Bricks.Backdrop - onClick() 2`] = `
<Backdrop
  background={false}
  className={null}
  controlled={true}
  disabled={false}
  hidden={false}
  id="idBackDrop"
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  onClick={[Function]}
  onMouseOver={[Function]}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
/>
`;

exports[`UU5.Bricks.Backdrop  props.function  UU5.Bricks.Backdrop - onMouseOver() 1`] = `
<Backdrop
  background={false}
  className={null}
  controlled={true}
  disabled={false}
  hidden={false}
  id="idBackDrop"
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  onClick={[Function]}
  onMouseOver={[Function]}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
/>
`;

exports[`UU5.Bricks.Backdrop  props.function  UU5.Bricks.Backdrop - onMouseOver() 2`] = `
<Backdrop
  background={false}
  className={null}
  controlled={true}
  disabled={false}
  hidden={false}
  id="idBackDrop"
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  onClick={[Function]}
  onMouseOver={[Function]}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
/>
`;

exports[`UU5.Bricks.Backdrop background 1`] = `
<div
  className="uu5-bricks-backdrop uu5-bricks-backdrop-background"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop background 2`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop background 3`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop background 4`] = `
<div
  className="uu5-bricks-backdrop uu5-bricks-backdrop-background"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop className 1`] = `
<div
  className="uu5-bricks-backdrop my-classname"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop controlled 1`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop controlled 2`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop controlled 3`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop controlled 4`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop default props 1`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop disabled 1`] = `
<div
  className="uu5-bricks-backdrop uu5-common-disabled"
  id="uuID"
  style={Object {}}
  title={null}
>
  <span
    className="uu5-common-disabled-cover"
    key="disabledCover"
  />
</div>
`;

exports[`UU5.Bricks.Backdrop disabled 2`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop disabled 3`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop disabled 4`] = `
<div
  className="uu5-bricks-backdrop uu5-common-disabled"
  id="uuID"
  style={Object {}}
  title={null}
>
  <span
    className="uu5-common-disabled-cover"
    key="disabledCover"
  />
</div>
`;

exports[`UU5.Bricks.Backdrop hidden 1`] = `
<div
  className="uu5-bricks-backdrop uu5-common-hidden"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop hidden 2`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop hidden 3`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop hidden 4`] = `
<div
  className="uu5-bricks-backdrop uu5-common-hidden"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop id 1`] = `
<div
  className="uu5-bricks-backdrop"
  id="myId"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop mainAttrs 1`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop name 1`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  name="myName"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 1`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 2`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 3`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 4`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 5`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 6`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 7`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 8`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 9`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 10`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 11`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 12`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 13`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 14`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 15`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 16`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 17`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 18`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop nestingLevel 19`] = `""`;

exports[`UU5.Bricks.Backdrop nestingLevel 20`] = `""`;

exports[`UU5.Bricks.Backdrop noIndex 1`] = `
<div
  className="uu5-bricks-backdrop uu5-noindex"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop noIndex 2`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop noIndex 3`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop noIndex 4`] = `
<div
  className="uu5-bricks-backdrop uu5-noindex"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop pureRender 1`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop pureRender 2`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop pureRender 3`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop pureRender 4`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop selected 1`] = `
<div
  className="uu5-bricks-backdrop uu5-common-selected"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop selected 2`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop selected 3`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop selected 4`] = `
<div
  className="uu5-bricks-backdrop uu5-common-selected"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop style 1`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop style 2`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop style 3`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop style 4`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Backdrop tooltip 1`] = `
<div
  className="uu5-bricks-backdrop"
  id="uuID"
  style={Object {}}
  title="My Tooltip"
/>
`;
